### PR TITLE
improve utility function and consolidate test

### PR DESCRIPTION
**Pull request by hankunxiao**

### CODE REVIEW

File: src/sharpedge/_utils/utility.py [hunk 1]
```diff
@@ -16,21 +16,20 @@
...
+        # Check the image array format: must be a numpy array
         if not isinstance(img_array, np.ndarray):
             raise TypeError("Image format must be a numpy array.")
         # Check input image array shape
         if not (img_array.ndim == 2 or (img_array.ndim == 3 and img_array.shape[2] == 3)):
-            raise ValueError(
-            f"Invalid image shape: {img_array.shape}. "
-            "Expected 2D (grayscale) or 3D with 3 channels (RGB)."
-            )
+            raise ValueError("Image array must be 2D or 3D.")
```
Comment (JennyonOort) [inline src/sharpedge/_utils/utility.py:24]: do you mind changing the msg into:
```suggestion
            raise ValueError("Image array must be 2D grayscale or 3D RGB.")
```
This is because we have other individual functions that use grayscale and RGB terms and we would want these terms to be used consistently across all functions.

File: tests/test_input_checker.py
Comment (JennyonOort) [inline tests/test_input_checker.py:1]: I am getting quite confused with the final consolidation of this.

- why are there now five parts instead of four? Why
- what happened to the edge cases? I know I had them.
- It seems the number of tests cases rhink by a lot. What are the reasons for that?

In general, I would suggest to keep to the maximum each person's  work in whole pieces so that it is easily recognizable or fixable going forward.

**Review by JennyonOort — COMMENTED** [review]
I am surprised by how short the test files are. I recall it was quite long when I tried to merge my part.

File: src/sharpedge/_utils/utility.py (hunk 1 above)
Comment (hankunxiao) [reply src/sharpedge/_utils/utility.py:24]: Yes! that's a good point.

File: tests/test_input_checker.py
Comment (hankunxiao) [reply tests/test_input_checker.py:1]: Hi Jenny, thanks for raising your concern.
The logic of our old condition check was not quite accurate and I update them. this is why you can find the 5 parts.
Since we don't raise any warning to be raised in function, we don't need to include any edge cases.
I appreciate everyone's work in compiling the test case, which helped me a lot when consolidating the test. 
You noticed a significant amount of test reduced this is mainly because:
1. I merged the redundant part of `expected cases`
2. The `edge cases` were removed
3. Some of the `erroneous cases` were not accurately and I removed them.

Comment (Arc-Celt) [reply tests/test_input_checker.py:1]: Just took a quick look at the updated `test_input_checker`. The organized version of the expected test cases looks much better now. Since we have the input checker as a whole, we don't need to design separate expected test cases for each error type, as those would be redundant.

As for the edge cases, I think we should definitely include them, since they're part of the test categories. (we don't want to lose points for missing those)

It's a bit tricky to spot exactly what changes have been made to the error cases, but as long as we have enough coverage for the different scenarios, it should be fine.

Comment (hankunxiao) [reply tests/test_input_checker.py:1]: thanks @Arc-Celt for your feedback I am adding some edge cases testing.

Comment (hankunxiao) [reply tests/test_input_checker.py:1]: @Arc-Celt I have included the edge cases in the latest commit

**Review by Arc-Celt — APPROVED** [review]
In order to accelerate our work progress, I think the current cases should be enough. The input checker is not the core functionality to test on. 

Let's wait for Jenny to review the latest test cases before proceeding with the merge though.

**Comment by Arc-Celt** [comment]
I’m not adding the input checker to the seam carving function right now because enforcing the 0–255 color range could cause inconsistent output results. (this happens because mapping the array from floating point to integers could lead to different energy compucations)

However, I [included other input validation test cases](https://github.com/UBC-MDS/SharpEdge/pull/65/files#diff-3917a1f52dc59a2cc3dd5a1b85a4aee2acaa8e7637ff5b2b9f1b8b52ec59f771R260-R266) (excluding the color range check, and I had the [dimension checks](https://github.com/UBC-MDS/SharpEdge/pull/65/files#diff-3917a1f52dc59a2cc3dd5a1b85a4aee2acaa8e7637ff5b2b9f1b8b52ec59f771R268-R270) already) in the seam carving tests later to keep input handling solid without affecting the core functionality. Feel free to check and bring up suggestions!
